src/test/kotlin/no/nav/syfo/tilgang/TilgangServiceSpek.kt

```diff
@@ -174,6 +175,173 @@ class TilgangServiceSpek : Spek({
+                val personidentSkjermet = Personident(UserConstants.PERSONIDENT_GRADERT)
+                val personidenter = listOf(personident.value, personidentSkjermet.value)
+                val cacheKeyAccess = "tilgang-til-person-${UserConstants.VEILEDER_IDENT}-$personident"
+                val cacheKeySkjermet = "tilgang-til-person-${UserConstants.VEILEDER_IDENT}-$personidentSkjermet"
+                every { redisStore.getObject<Tilgang?>(any()) } returns null
+                coEvery { graphApiClient.hasAccess(adRoller.SYFO, any(), any()) } returns true
+                coEvery { behandlendeEnhetClient.getEnhetWithOboToken(any(), personident, any()) } returns behandlendeEnhet
+                coEvery { behandlendeEnhetClient.getEnhetWithOboToken(any(), personidentSkjermet, any()) } returns behandlendeEnhet
+                coEvery { axsysClient.getEnheter(any(), any()) } returns listOf(veiledersEnhet)
+                coEvery { skjermedePersonerPipClient.getIsSkjermetWithOboToken(any(), personident, any()) } returns false
+                coEvery { skjermedePersonerPipClient.getIsSkjermetWithOboToken(any(), personidentSkjermet, any()) } returns true
+                coEvery { pdlClient.getPersonWithOboToken(any(), personident, any()) } returns ugradertInnbygger
+                coEvery { graphApiClient.hasAccess(adRoller.EGEN_ANSATT, any(), any()) } returns false
+
+                runBlocking {
+                    val filteredPersonidenter = tilgangService.filterIdenterByVeilederAccess(
+                        token = validToken,
+                        callId = callId,
+                        personidenter = personidenter,
+                    )
+
+                    filteredPersonidenter.size shouldBeEqualTo 1
+                    filteredPersonidenter[0] shouldBeEqualTo UserConstants.PERSONIDENT
+                }
+
+                coVerify(exactly = 2) { graphApiClient.hasAccess(adRoller.SYFO, validToken, callId) }
+                coVerify(exactly = 1) { behandlendeEnhetClient.getEnhetWithOboToken(callId, personident, validToken) }
+                coVerify(exactly = 1) { behandlendeEnhetClient.getEnhetWithOboToken(callId, personidentSkjermet, validToken) }
+                coVerify(exactly = 1) { skjermedePersonerPipClient.getIsSkjermetWithOboToken(callId, personident, validToken) }
+                coVerify(exactly = 1) { skjermedePersonerPipClient.getIsSkjermetWithOboToken(callId, personidentSkjermet, validToken) }
+                coVerify(exactly = 1) { pdlClient.getPersonWithOboToken(callId, personident, validToken) }
+                coVerify(exactly = 0) { pdlClient.getPersonWithOboToken(any(), personidentSkjermet, any()) }
+                verifyCacheSet(exactly = 1, key = cacheKeyAccess, harTilgang = true)
+                verifyCacheSet(exactly = 1, key = cacheKeySkjermet, harTilgang = false)
+            }
+
+            it("remove innbygger when veileder is missing correct access") {
```

Comment:
Kjempefin test! 🙈 Den fjerner en person for hvert "steg" i tilgangssjekken - geografisk, skjerming, adressebeskytelse. Man kunne evt. bare lagd én test per, sånn at det i tillegg til "skjermet"-testen over, får man én test for geografisk og én for kode6/7, og alle verifiserer at personlisten går fra 2 til 1 element.

Ellers er det jo egentlig samme kode som for persontilgang da, så jeg vet ikke hvor nøye det er 🤷‍♂️ 

Noe av årsaken til at dette blir så langt, er at personident er en `value class` og de funker ikke med coEvery, så man får ikke brukt `any()`, men må lage en mock per personident. (I coVerify blir det nok én rad per uansett.)

Reply:
Leit at `value class` skulle være vanskelig da :/